import_export/static/import_export/export_selectable_fields.js

```diff
@@ -0,0 +1,45 @@
+function hideUnselectedResourceFields(selectedResourceIndex) {
+    const fields = document.querySelectorAll("[resource-index]")
+
+    fields.forEach(field => {
+        if (field.getAttribute("resource-index") !== selectedResourceIndex.toString()) {
+            // field is wrapped by div, change visibility on wrapper
+            field.style.display = "none"
+        }
+    })
+}
+
+function showSelectedResourceFields(resourceIndex) {
+    const fields = document.querySelectorAll(`[resource-index="${resourceIndex}"]`)
+
+    fields.forEach(field => {
+        // field is wrapped by div, change visibility on wrapper
+        field.style.display = "block"
+    })
+}
+
+function onResourceSelected(e) {
+    const resourceIndex = e.target.value
+
+    showSelectedResourceFields(resourceIndex)
+
+    hideUnselectedResourceFields(resourceIndex)
+}
+
+document.addEventListener("DOMContentLoaded", function() {
+    const resourceSelector = document.querySelector("#id_resource")
+
+    if (!resourceSelector) {
+        console.error("resource select input not found")
+        return
+    }
+
+    // If selector is actually select input, get selected option.
+    // else selected resource index is 0
+    const selectedResourceIndex = resourceSelector.tagName === "SELECT" ? resourceSelector.value : 0
+
+    resourceSelector.addEventListener("input", onResourceSelected)
+
+    // initially hide unselected resource fields
+    hideUnselectedResourceFields(selectedResourceIndex)
```

Comment:
I suggest add ';' to end of statements for best practice ([link](https://tc39.es/ecma262/#sec-automatic-semicolon-insertion)).  What do you think?

Reply:
If it helps, you can use jQuery instead of core javascript calls

Reply:
> I suggest add ';' to end of statements for best practice ([link](https://tc39.es/ecma262/#sec-automatic-semicolon-insertion)). What do you think?

I usually don't use semicolons with js/ts but I will refactor accordingly to project's standards.



> If it helps, you can use jQuery instead of core javascript calls

It's been long time since I have used jQuery, there is nothing fancy going that *would* cause compatibility issue, so I prefer JS.